victorops api stuff

Page- opened the pull request.

Page- committed:
victorops api stuff

Page- committed:
improvements

Page- commented on get-victorops-schedule.ts:
```diff
@@ -0,0 +1,261 @@
+import * as _ from 'lodash';
+import { google, calendar_v3 } from 'googleapis';
+import { getAuthClient } from './lib/gauth';
+import { agents } from './logs/2022-08-01_devOps/support-shift-scheduler-input.json';
```
This needs to be updated to point to an actually existing file, and it needs to have entries for every person that might be part of the victorops schedule (or otherwise it'll bork saying that an attendee doesn't have an email)

cmfcruz committed:
Update VO schedule script: add dry-run mode, update auth, and add missing dependency

- Add DRY_RUN env var to preview calendar events without creating them
- Update gauth import from removed getAuthClient to getJWTAuthClient
- Increase victorops-api-client maxContentLength to handle larger responses
- Add date-fns as an explicit dependency
- Update input JSON path to current schedule period

Change-type: patch
Signed-off-by: Carlo Miguel F. Cruz <carloc@balena.io>

cmfcruz committed:
Rename helper script and add npm run script

- Renames script to match function
- Adds an equivalent npm run script for helper script consistency

Change-type: patch
Signed-off-by: Carlo Miguel F. Cruz <carloc@balena.io>

AlidaOdendaal commented on helper-scripts/set-devops-weekend-schedule.ts:
```diff
@@ -0,0 +1,316 @@
+const DAY_SKIP_MAX = 90;
+const DAYS_FORWARD_MAX = 123;
+async function loadHandleToEmail(inputDate: string) {
+	const inputFile = `logs/${inputDate}_${scheduleName}/support-shift-scheduler-input.json`;
```
I want to suggest that, instead of relying on a different tab of "devOps Support Scheduler Logs" every time to get the github handle -> e-mail map, we pull that data, as well as the VictorOps usernames, from [this](https://docs.google.com/spreadsheets/d/1RsQjoMqQaw2Xv_Dk6UobmsBSnNJy2cGuKE57C2qIo2s/edit?gid=1389048137#gid=1389048137) tab I've created in the Teamwork Model. Then we also refactor the weekday DevOps scheduling to make use of that tab, and remove the `victoropsUsernames` from `helper-scripts/options/devOps.json`.

AlidaOdendaal commented on helper-scripts/set-devops-weekend-schedule.ts:
```diff
@@ -0,0 +1,316 @@
+
+	const until = end;
+
+	while (end <= until) {
```
I believe this `while` loop will only ever execute once? Since `until` is set to `end` just before the loop, and `end` is incremented by 7 days at the bottom of the loop. Was the intention perhaps to break up the API calls into 7-day increments? Then we could introduce e.g. `windowStart` and `windowEnd` variables to demarcate the chunks.

cmfcruz replied:
I removed the misleading while loop.  We only need to execute the script once to retrieve the schedules from VictorOps.

AlidaOdendaal commented on helper-scripts/set-devops-weekend-schedule.ts:
```diff
@@ -0,0 +1,316 @@
+						sendUpdates: 'all',
+						requestBody: eventResource,
+					});
+				}
```
Since this event definition and insertion is the same as in `futureWeekends`, it would be good to define it as a reusable function instead.

cmfcruz replied:
I applied some changes for this to de-duplicate the code.

cmfcruz committed:
Simplify past DevOps weekend schedule fetch

Remove the misleading weekly loop from the past schedule path. The loop
only executed once because `until` was initialized from `end`, so fetching
the requested VictorOps date range directly matches the current behavior
while making the intent clearer.

Change-type: patch
Signed-off-by: Carlo Miguel F. Cruz <carloc@balena.io>

cmfcruz committed:
Deduplicate DevOps weekend calendar event insertion

Change-type: patch
Signed-off-by: Carlo Miguel F. Cruz <carloc@balena.io>

AlidaOdendaal reviewed (approved):
Looks great, thanks @cmfcruz !
